### DISCUSSION TITLE
Pedalboard

### DISCUSSION THREAD

**Discussion by its-javierjimenez** [flet-dev/flet#4288] [Packages] [answered]
https://github.com/spotify/pedalboard

**Comment by ndonkoHenri** [comment, marked as answer]
Spotify's pedalboard can't be packaged for Flet mobile:

- No source distribution on PyPi, so there's nothing for Flet's mobile build to cross-compile.
- It bundles the full JUCE framework, including its desktop GUI modules (`juce_gui_basics`/`juce_graphics`/`juce_gui_extra`), which assume a windowing system (Cocoa/X11) and pull desktop system libs (X11, freetype, ALSA). These would have to be stripped/stubbed just to build.
- Its headline feature — VST3/AU plugin hosting — dlopens native plugin binaries at runtime, which don't exist and can't be loaded on iOS/Android. (License is also GPLv3)

A possible alternative could be to run pedalboard on a backend service and stream the result to the app.
